api/v1/swagger/paths/holdings.json

```diff
@@ -0,0 +1,352 @@
+          "type": "string"
+        },
+        {
+          "name": "format",
+          "in": "query",
+          "description": "Case insensitive starts-with search for format",
+          "type": "string"
+        },
+        {
+          "name": "deleted_on",
+          "in": "query",
+          "description": "Case insensitive starts-with search for date this record was deleted from Koha",
+          "type": "string"
+        }
+      ],
+      "produces": ["application/json"],
+      "responses": {
+        "200": {
+          "description": "A list of holdings",
+          "schema": {
+            "$ref": "../definitions.json#/holdings"
+          }
+        },
+        "404": {
+          "description": "No holdings found",
+          "schema": {
+            "$ref": "../definitions.json#/error"
+          }
+        },
+        "500": {
+          "description": "Internal server error",
+          "schema": {
+            "$ref": "../definitions.json#/error"
+          }
+        },
+        "503": {
+          "description": "Under maintenance",
+          "schema": {
+            "$ref": "../definitions.json#/error"
+          }
+        }
+      }
+    },
+    "post": {
+      "x-mojo-to": "Holding#add",
+      "operationId": "addHolding",
+      "tags": ["patrons", "holdings"],
+      "parameters": [{
+        "name": "body",
+        "in": "body",
+        "description": "A JSON object containing informations about the new holding",
+        "required": true,
+        "schema": {
+          "$ref": "../definitions.json#/holding"
+        }
+      }
+      ],
+      "consumes": ["application/json"],
+      "produces": ["application/json"],
+      "responses": {
+        "201": {
+          "description": "Created holding",
+          "schema": {
+            "$ref": "../definitions.json#/holding"
+          }
+        },
+        "400": {
+          "description": "Missing or wrong parameters",
+          "schema": {
+            "$ref": "../definitions.json#/error"
+          }
+        },
+        "401": {
+          "description": "Authentication required",
+          "schema": {
+            "$ref": "../definitions.json#/error"
+          }
+        },
+        "403": {
+          "description": "Holding not allowed",
+          "schema": {
+            "$ref": "../definitions.json#/error"
+          }
+        },
+        "404": {
+          "description": "Borrower not found",
+          "schema": {
+            "$ref": "../definitions.json#/error"
+          }
+        },
+        "500": {
+          "description": "Internal server error",
+          "schema": {
+            "$ref": "../definitions.json#/error"
+          }
+        },
+        "503": {
+          "description": "Under maintenance",
+          "schema": {
+            "$ref": "../definitions.json#/error"
+          }
+        }
+      },
+      "x-koha-authorization": {
+        "permissions": {
+          "editcatalogue": "add_catalogue"
```

Comment:
You should add a new set of permissions for holdings CRUD

{holdings => qw(create read update delete)}

It would be better to implement those in the staff client as well, but not really imho.

Reply:
Adding new permissions is rather easy.

Reply:
The edit_catalogue permission in Koha is truly too large.
It might be desirable to have a specific client to just update holdings-records. eg an acquisitions extension.

Reply:
@kivilahtio I don't know if this is time usefully spent since KS and NatLibFi are moving to KC Koha where the permissions are differently implemented. IMHO, let's just use the add_catalogue. and also because then we should implement separate "add_catalogue" permission for items and not just holdings records, so this is starting to feel like out of scope.

Reply:
Personally I'm fine either way. If (the sponsor) NatLibFi thinks it's not worth it, I'll mark this resolved.

Reply:
In fact we already have a separate "edit_items" permission (but not add or delete). So, it's not actually a so far fetched idea to add them for holdings.